test/unit/ngsiv2/provisioning/singleConfigurationMode-test.js

```diff
@@ -0,0 +1,333 @@
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            request(groupCreation, function(error) {
+                request(deviceCreation, function(error, response, body) {
+                    done();
+                });
+            });
+        });
+
+        it('should raise a DUPLICATE_DEVICE_ID error', function(done) {
+            request(deviceCreationDuplicated, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(409);
+                should.exist(body.name);
+                body.name.should.equal('DUPLICATE_DEVICE_ID');
+                done();
+            });
+        });
+    });
+    describe('When a device is provisioned with an ID that exists globally but not in the configuration', function() {
+        var alternativeDeviceCreation = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+                method: 'POST',
+                json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionNewDevice.json'),
+                headers: {
+                    'fiware-service': 'AlternateService',
+                    'fiware-servicepath': '/testingPath'
+                }
+            },
+            alternativeGroupCreation = {
+                url: 'http://localhost:4041/iot/services',
+                method: 'POST',
+                json: utils.readExampleFile('./test/unit/examples/groupProvisioningRequests/provisionFullGroup.json'),
+                headers: {
+                    'fiware-service': 'AlternateService',
+                    'fiware-servicepath': '/testingPath'
+                }
+            };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'AlternateService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'AlternateService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5